sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java

```diff
@@ -476,19 +483,26 @@ public TableClient createTableIfNotExists(String tableName) {
     public Response<TableClient> createTableIfNotExistsWithResponse(String tableName, Duration timeout,
                                                                     Context context) {
         Supplier<Response<TableClient>> callable = () -> createTableIfNotExistsWithResponse(tableName, context);
-        return callWithOptionalTimeout(callable, THREAD_POOL, timeout, logger);
+        try {
+            return callWithOptionalTimeout(callable, THREAD_POOL, timeout);
+        } catch (InterruptedException | ExecutionException | TimeoutException ex) {
+            throw logger.logExceptionAsError(new RuntimeException(ex));
+        } catch (RuntimeException ex) {
+            throw logger.logExceptionAsError((RuntimeException) mapThrowableToTableServiceException(ex));
+        }
     }
 
     Response<TableClient> createTableIfNotExistsWithResponse(String tableName, Context context) {
         try {
             return createTableWithResponse(tableName, null, context);
         } catch (Exception e) {
-            if (e instanceof TableServiceException
-                && ((TableServiceException) e).getResponse() != null
-                && ((TableServiceException) e).getResponse().getStatusCode() == 409) {
-                HttpResponse response = ((TableServiceException) e).getResponse();
-                return new SimpleResponse<>(response.getRequest(), response.getStatusCode(),
-                response.getHeaders(), null);
+            if (mapThrowableToTableServiceException(e) instanceof TableServiceException) {
+                TableServiceException exception = (TableServiceException) mapThrowableToTableServiceException(e);
+                if (exception.getResponse() != null && exception.getResponse().getStatusCode() == 409) {
+                    HttpResponse response = exception.getResponse();
+                    return new SimpleResponse<>(response.getRequest(), response.getStatusCode(),
+                        response.getHeaders(), null);
+                }
```

Comment:
This doesn't fix the logging problem, it's the call to `createTableWithResponse` that ends up logging when it gets back a 409/412 exception.

This method, and the one in the async client, will have to duplicate the logic of `createTableWithResponse` without the logging when a 409/412 is returned

Reply:
@alzimmermsft I accidentally undid my changes 😆. But I was thinking that instead of duplicating logic that `context` could be used to decide when to log or not.

Reply:
Might be better to create a shared method that doesn't catch the thrown exception and leave it to create and createIfNotExists to manage what is logged

Reply:
I think that makes sense

Reply:
@alzimmermsft The async clients do not have this logging issue. The logging of an error message only happens in the sync client.